contracts/Clearable.sol

```diff
@@ -0,0 +1,159 @@
+pragma solidity ^0.5.0;
+
+import "eip1996/contracts/Holdable.sol";
+import "./IClearable.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+contract Clearable is Holdable, IClearable, Ownable {
+
+    using StringUtil for string;
+
+    struct ClearableTransfer {
+        address orderer;
+        address from;
+        address to;
+        address clearableAgent;
+        uint256 value;
+        ClearableTransferStatusCode status;
+    }
+
+    mapping(bytes32 => ClearableTransfer) internal clearableTransfers;
+    mapping(address => mapping(address => bool)) private operators;
+    address clearableAgent;
+
+    constructor() public{
+        clearableAgent = msg.sender; 
+    }
+
+    function orderTransfer(string calldata operationId, address to, uint256 value) external returns (bool) {
+        require(to != address(0), "Payee address must not be zero address");
+        return _orderTransfer(
+            operationId,
+            msg.sender,
+            msg.sender,
+            to,
+            value
+            );
+    }
+
+    function orderTransferFrom(string calldata operationId, address from, address to, uint256 value) external returns (bool) {
+        require(to != address(0), "Payee address must not be zero address");
+        require(from != address(0), "Payer address must not be zero address");
+        require(operators[from][msg.sender] == true, "This operator is not authorized");
+        return _orderTransfer(
+            operationId,
+            msg.sender,
+            from,
+            to,
+            value
+            );
+    }
+
+    function cancelTransfer(string calldata operationId) external returns (bool) {
+        ClearableTransfer storage newClearableTransfer = clearableTransfers[operationId.toHash()];
+        require (msg.sender == newClearableTransfer.from, "Can only be processed by the payer");
```

Comment:
To get the `from` from the hold you can do:
```
Hold storage newClearableHold = holds[operationId.toHash()];
require (msg.sender == newClearableHold.from, "Can only be processed by the payer");
```

Reply:
Ok but in the hold struct we have origin and target is better from and to or doesn't it matter?